CMakeLists.txt

```diff
@@ -127,9 +127,12 @@ find_package(ZLIB REQUIRED)
 
 if(NOT TARGET CURL::CURL)
   add_library(CURL::CURL INTERFACE IMPORTED)
-  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS)
+  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS OR ENABLE_PYTHON_BINDINGS)
     if(NOT CURL_FOUND)
       find_package(CURL REQUIRED)
+      if (WIN32 AND ENABLE_PYTHON_BINDINGS)
+        link_libraries(${CURL_LIBRARIES})
```

Comment:
im guess on windows you are getting a linker error without this? im wondering why that would be.. surely pybind11 doesnt need curl linking?

Reply:
ahh i see the discussion above, can you post the link error you get when you comment out these lines. maybe somethign will pop out at someone

Reply:
jup, linker error. pybind doesn't need curl, must be some include in `python.cc`, couldn't really decipher which one though. it was the same with boost.python.

Reply:
@nilsnolde If you run CMake with Visual Studio generator and open the solution in VS IDE, you can edit properties of the bindings project, and toggle the [/showIncludes](https://docs.microsoft.com/en-us/cpp/build/reference/showincludes-list-include-files), then build just the bindings (or just compile the `python.cc`) and it will dump all the headers included from the `python.cc`. That may give some clues.

Reply:
I tried, but couldn't make it work.. Always worked in VS Code, I couldn't stand VS IDE from the first moment on.. I tried a `new project from existing sources` and (with a minute-old fresh install), it configured alright, but then the properties are grayed out:
![image](https://user-images.githubusercontent.com/25637358/95977896-b941ca80-0e19-11eb-9eff-1b88b67a100c.png)

And under Project menu item there's no Properties entry either. 

For @kevinkreiser, here's the error log, unable to link to curl:
```
[build]      Creating library C:/Users/nilsn/Documents/dev/cpp/valhalla/build/src/bindings/python/Release/python_valhalla.lib and object C:/Users/nilsn/Documents/dev/cpp/valhalla/build/src/bindings/python/Release/python_valhalla.exp
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_global_init [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_global_cleanup [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_easy_init [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_easy_setopt [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_easy_perform [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_easy_cleanup [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] valhalla.lib(curler.obj) : error LNK2001: unresolved external symbol __imp_curl_easy_getinfo [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
[build] C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\Release\python_valhalla.cp39-win_amd64.pyd : fatal error LNK1120: 7 unresolved externals [C:\Users\nilsn\Documents\dev\cpp\valhalla\build\src\bindings\python\python_valhalla.vcxproj]
```

Reply:
@nilsnolde That is because you opened CMake directly, so VS runs in CMake mode. I suggested to run CMake from CLI to generate `.sln` and `.vcxproj` files, what then allows you to compile single source files.
Anyway, it does not have to be VS IDE, it can be any UI you prefer, CLI included, as long as you can add `/showIncludes` to `CL.EXE` command line, what actually was the gist of my suggestion.

Reply:
when you are configuring are you configuring with `ENABLE_HTTP=On`? if not, ie if you have it set to `Off` (the default is `On`), then the code inside curler that makes use of curl should be disabled unless for some reason we have python bindings turning ENABLE_HTTP back on? we should probably have a look through cmake to see if there is a bug there somewhere...

Reply:
Pretty sure it's not exactly what you were after, anyways here's the output of compiling python.cc until it can't find the proto headers (which need to be compiled first): https://gist.github.com/nilsnolde/3d7d81d762e1dc21a9f97e70eba96848

I couldn't follow really what you were saying.. I have no idea how to use `.vcxproj` files with `cl.exe`.. The above is the output simply linking a few headers before compiling python.cc in the `valhalla/src/bindings/python/` folder

Reply:
Apparently it doesn't matter for the MVSC build whether ENABLE_HTTP is ON or OFF @kevinkreiser. Both configs fail in the exact same way when PYTHON_BINDINGS is ON (without the extra lines above).

Reply:
very interesting we must have a long standing bug there then? ill have a look later on today!

Reply:
thanks @mloskot :)

Reply:
is it possible that the python dll in windows links curl but somehow doesnt advertise that fact? on my version of linux libpython does not link curl and ive had a look at the python soruce and its configure.ac also doesnt seem to reference curl as a dependency. that makes me think that indeed we have some other long standing bug in our cmake files that is causing windows to require curl all the time! still didnt get a chance to look yet but i'll do so shortly. imho this little oddity is the only thing blockign this PR

Reply:
Oh boy, now it gets weird.. Python is not dependent on curl: When I compile with `ENABLE_DATA_TOOLS=OFF` (and ENABLE_HTTP=OFF), I have no issues with curl and the python target (and all others which are enabled in that combo) build smoothly. 

It seems smth is not working right with `curler.cc` `#ifdef CURL_STATICLIB`.. If I switch that with smth I know I don't have (e.g. `#ifdef HAVE_HTTP=ON`), then all targets compile **also with `ENABLE_DATA_TOOLS=ON`**. There's even targets in `DATA_TOOLS` which use curl, e.g. `valhalla_fetch_transit` and running that doesn't seem to be an issue:
```
2020/10/15 08:20:04.872656 ←[32;1m[INFO]←[0m Fetching transit feeds
2020/10/15 08:20:07.601253 ←[33;1m[WARN]←[0m 404'd retrying http://transit.land//api/v1/feeds.geojson?per_page=false&api_key=transitland-YOUR_KEY_SUFFIX&bbox=-122.469,37.502,-121.78,38.018&active_feed_version_import_level=4
```

So smth weird seems to happen with the combo Python/ENABLE_DATA_TOOLS and libcurl. Does that ring a bell with anyone?